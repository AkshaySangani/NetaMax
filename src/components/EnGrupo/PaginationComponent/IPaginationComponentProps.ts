export interface IPaginationComponentProps {
  /**
   * Page for accordion.
   * @type {number}
   */
  page: number;

  /**
   * TotalPage for accordion.
   * @type {number}
   */
  totalPage: number;

  /**
   * OnChangePage for accordion.
   * @type {(page: number) => void}
   */
  onChangePage: (page: number) => void;
}
