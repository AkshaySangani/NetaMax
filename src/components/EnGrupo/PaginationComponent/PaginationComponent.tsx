import { ReactElement } from "react";

import { Box, Text } from "@chakra-ui/react";

import { IPaginationComponentProps } from "./IPaginationComponentProps";

/**
 * The PaginationComponent component.
 * @param {IPaginationComponentProps} props the component props.
 * @returns {ReactElement} the component element.
 */
const PaginationComponent = (props: IPaginationComponentProps): ReactElement => {
  const { page, totalPage, onChangePage } = props;
  const totalPageList = Array.from({ length: totalPage }, (_, i) => i + 1);

  return (
    <Box
      display="flex"
      flexDirection="row"
      justifyContent="center"
      alignItems="center"
      padding="1rem"
    >
      {totalPageList.map((pageItem, index) => {
        const pageItemActive = pageItem === page;

        const itemPaginationActive = pageItemActive
          ? {
              backgroundColor: "#353535",
              borderRadius: "50%",
            }
          : {};
        const textPaginationActive = pageItemActive
          ? {
              textColor: "#FFFFFF",
            }
          : {};

        return (
          <Box
            key={index}
            width={35}
            height={35}
            margin="0 0.5rem"
            display="flex"
            flexDirection="column"
            justifyContent="center"
            alignItems="center"
            cursor="pointer"
            {...itemPaginationActive}
            onClick={() => onChangePage(pageItem)}
          >
            <Text textColor="#242424" fontWeight={600} fontSize={16} {...textPaginationActive}>
              {pageItem}
            </Text>
          </Box>
        );
      })}
    </Box>
  );
};

export default PaginationComponent;
