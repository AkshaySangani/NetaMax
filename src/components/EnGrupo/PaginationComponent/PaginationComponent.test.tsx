import { render, fireEvent } from "@testing-library/react";
import PaginationComponent from "./PaginationComponent";

jest.mock("next/router", () => require("next-router-mock"));

describe("Render PaginationComponent component", () => {
  it("PaginationComponent should be render correctly", () => {
    const { container } = render(
      <PaginationComponent page={1} totalPage={2} onChangePage={() => null} />
    );
    expect(container).toMatchSnapshot();
  });

  it("PaginationComponent should click in page", () => {
    const onClickMock = jest.fn();
    const { getByText } = render(
      <PaginationComponent page={1} totalPage={2} onChangePage={onClickMock} />
    );
    fireEvent.click(getByText("1"));
    expect(onClickMock).toBeCalledTimes(1);
  });
});
