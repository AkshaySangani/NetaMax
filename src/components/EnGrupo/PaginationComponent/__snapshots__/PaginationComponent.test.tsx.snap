// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render PaginationComponent component PaginationComponent should be render correctly 1`] = `
<div>
  <div
    class="css-o4syeq"
  >
    <div
      class="css-6u6xpa"
    >
      <p
        class="chakra-text css-1x4je9a"
      >
        1
      </p>
    </div>
    <div
      class="css-sg6es6"
    >
      <p
        class="chakra-text css-qc0huw"
      >
        2
      </p>
    </div>
  </div>
</div>
`;
